use-cases/xtclim/src/anomaly.py

```diff
@@ -1,114 +1,182 @@
+            train_time = pd.read_csv(f"input/dates_train_{season}data_1memb.csv")
+            train_data = np.load(f"input/preprocessed_1d_train_{season}data_1memb.npy")
+            n_train = len(train_data)
+            trainset = [ ( torch.from_numpy(np.reshape(train_data[i], (2, 32, 32))),
+                    train_time['0'][i] ) for i in range(n_train) ]
+            trainloader = self.strategy.create_dataloader(
+                trainset, batch_size=self.config.batch_size, shuffle=False
+            )
+
+            # test set and data loader
+            test_time = pd.read_csv(f"input/dates_test_{season}data_1memb.csv")
+            test_data = np.load(f"input/preprocessed_1d_test_{season}data_1memb.npy")
+            n_test = len(test_data)
+            testset = [ ( torch.from_numpy(np.reshape(test_data[i], (2, 32, 32))),
+                    test_time['0'][i] ) for i in range(n_test) ]
+            testloader = self.strategy.create_dataloader(
+                testset, batch_size=self.config.batch_size, shuffle=False
+            )
+
+            if self.strategy.is_main_worker and self.logger:
+                self.logger.create_logger_context()
+
+            # average over a few iterations
+            # for a better reconstruction estimate
+            train_avg_losses, _, tot_train_losses, _ = evaluate(cvae_model, trainloader,
+                                                     trainset, self.device,
+                                                    criterion,
+                                                    pixel_wise_criterion)
+            test_avg_losses, _, tot_test_losses, _ = evaluate(cvae_model, testloader,
+                                                   testset, self.device, criterion,
+                                                   pixel_wise_criterion)
             for i in range(1, n_avg):
-                proj_avg_loss, _, proj_losses, _ = evaluate(cvae_model, projloader, 
-                                                        projset, device, criterion, 
-                                                        pixel_wise_criterion)
-                tot_proj_losses = list(map(add, tot_proj_losses, proj_losses))
-                proj_avg_losses += proj_avg_loss
-            
-            tot_proj_losses = np.array(tot_proj_losses)/n_avg
-            proj_avg_losses = proj_avg_losses/n_avg
-
-            # save the losses time series
-            pd.DataFrame(tot_proj_losses).to_csv(f"../outputs/proj{scenario}_losses_{season}1d_allssp.csv")
-            print(f'SSP{scenario} Projection average loss:', proj_avg_losses, 'for', season[:-1])
\ No newline at end of file
+                train_avg_loss, _, train_losses, _ = evaluate(cvae_model, trainloader,
+                                                 trainset, self.device, criterion,
+                                                 pixel_wise_criterion)
+                tot_train_losses = list(map(add, tot_train_losses, train_losses))
```

Comment:
Generally, one wants to avoid the "map" function in Python as it is not very "Pythonic" (ironically, since it is built in). Usually, you'd rather use some form of list comprehension or in this case, since you are doing element-wise addition, use numpy.